source/common/network/connection_impl.cc

```diff
@@ -780,5 +780,1079 @@ void ClientConnectionImpl::connect() {
     }
   }
 }
+
+ClientPipeImpl::ClientPipeImpl(Event::Dispatcher& dispatcher,
```

Comment:
Please break this into its own file.

Reply:
ClientPipeImpl is now deleted
Instead of introducing ClientPipeImpl and ServerPipeImpl, `ConnectionImpl` now could use new buffer source io handle.